services/app/src/contexts/ws-context.jsx

```diff
@@ -59,6 +59,15 @@ const WebSocketsContextProvider = ({ children }) => {
     }
   }, [ws])
 
+  window.onfocus = () => {
+    if(ws?.readyState === wsReadyStates.CLOSED) {
+      const webSocket = openWsConnection(url);
+      webSocket.addEventListener('open', () => {
```

Comment:
There is code duplication here.

Update openWsConnection to accept `onOpen` callback.

e.g.
```
openWsConnection(url, {
   onOpen: ws => setWs(ws),
})
```